### PR TITLE
refactor(autoware_pointcloud_preprocessor): rework pointcloud accumulator parameters 

### PR DESCRIPTION
**Pull request by vividf**
## Description
This PR includes the following changes
1. Remove the default parameters for the pointcloud accumulator node. 
2. Add parameter, launch file, and parameters schema for the node. 
3. Change the name `nodelet` to `node`.

A big thank you to @Ariiees for the PRs (https://github.com/autowarefoundation/autoware.universe/pull/8297 and https://github.com/autowarefoundation/autoware.universe/pull/7422), where we worked together on the entire pointcloud preprocessor parameters, launch file, and schema.

## Related links

**Parent Issue:**

- Link

## How was this PR tested?
`ros2 launch autoware_pointcloud_preprocessor pointcloud_accumulator_node.launch.xml`
## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: sensing/autoware_pointcloud_preprocessor/src/pointcloud_accumulator/pointcloud_accumulator_node.cpp [hunk 1]
```diff
@@ -23,9 +23,8 @@ PointcloudAccumulatorComponent::PointcloudAccumulatorComponent(const rclcpp::Nod
 {
   // set initial parameters
   {
-    accumulation_time_sec_ = static_cast<double>(declare_parameter("accumulation_time_sec", 2.0));
-    pointcloud_buffer_.set_capacity(
-      static_cast<size_t>(declare_parameter("pointcloud_buffer_size", 50)));
+    accumulation_time_sec_ = declare_parameter<double>("accumulation_time_sec");
+    pointcloud_buffer_.set_capacity(declare_parameter<int64_t>("pointcloud_buffer_size"));
```
Comment (knzo25) [inline sensing/autoware_pointcloud_preprocessor/src/pointcloud_accumulator/pointcloud_accumulator_node.cpp:27]: The boost class uses size_t it seems
https://www.boost.org/doc/libs/1_42_0/libs/circular_buffer/doc/circular_buffer.html

**Review by knzo25 — CHANGES_REQUESTED** [review]
Same comment as in the related PRs

**Comment by knzo25** [comment]
@vividf 
If you have several related PRs, please leave a link to them.
Comments related to one PR in the series may apply to the others as well

File: sensing/autoware_pointcloud_preprocessor/src/pointcloud_accumulator/pointcloud_accumulator_node.cpp (hunk 1 above)
Comment (vividf) [reply sensing/autoware_pointcloud_preprocessor/src/pointcloud_accumulator/pointcloud_accumulator_node.cpp:27]: @knzo25 thanks! However, declare_parameter<size_t> doesn't work for ros2 parameter. (compile will fatiled)
image

reference: https://autowarefoundation.github.io/autoware-documentation/main/contributing/coding-guidelines/ros-nodes/parameters/

Comment (knzo25) [reply sensing/autoware_pointcloud_preprocessor/src/pointcloud_accumulator/pointcloud_accumulator_node.cpp:27]: *compile fails

I know that size_t is incompatible (internally it struggles with signedness). As of this PR there is an implicit cast which needs to made explicit. This applies to all other PRs

Comment (vividf) [reply sensing/autoware_pointcloud_preprocessor/src/pointcloud_accumulator/pointcloud_accumulator_node.cpp]: Thanks, fixed in https://github.com/autowarefoundation/autoware.universe/pull/8487/commits/e125de73b329a76be7c625ce1cf11a3fb01f2f16

**Comment by vividf** [comment]
PRs that need an additional cast for variables: 
1. https://github.com/autowarefoundation/autoware.universe/pull/8487
2. https://github.com/autowarefoundation/autoware.universe/pull/8485
3. https://github.com/autowarefoundation/autoware.universe/pull/8474
4. https://github.com/autowarefoundation/autoware.universe/pull/8469
5. https://github.com/autowarefoundation/autoware.universe/pull/8468

**Comment by vividf** [comment]
@knzo25 kindly ping

**Review by knzo25 — APPROVED** [review]
LGTM

**Review by knzo25 — APPROVED** [review]
LGTM

**Pull request merged**